File: common/src/main/java/org/agent/hexstruction/Hexstruction.java
package org.agent.hexstruction;

import at.petrak.hexcasting.common.lib.hex.HexIotaTypes;
import net.minecraft.core.Registry;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import net.minecraft.resources.ResourceLocation;

// todo: configure forge gradle
public final class Hexstruction {
    public static final String MOD_ID = "hexstruction";
    public static final Logger LOGGER = LogManager.getLogger(MOD_ID);

    public static void init() {
        LOGGER.info("Initializing HexStruction");
        HexStructionPatterns.Init();
        Registry.register(HexIotaTypes.REGISTRY, GetID("structure"), StructureIota.TYPE);
    }

    public static ResourceLocation GetID(String name) {
        return new ResourceLocation(MOD_ID, name);
    }
}

File: common/src/main/kotlin/org/agent/hexstruction/Hexstruction.kt
package org.agent.hexstruction

import net.minecraft.resources.ResourceLocation
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import org.agent.hexstruction.config.HexstructionConfig
import org.agent.hexstruction.networking.HexstructionNetworking
import org.agent.hexstruction.registry.HexstructionActions
import org.agent.hexstruction.registry.HexstructionIotaTypes

object Hexstruction {
    const val MODID = "hexstruction"

    @JvmField
    val LOGGER: Logger = LogManager.getLogger(MODID)

    @JvmStatic
    fun id(path: String) = ResourceLocation(MODID, path)

    fun init() {
        HexstructionConfig.init()
        initRegistries(
            HexstructionActions,
            HexstructionIotaTypes
        )
        HexstructionNetworking.init()
    }
}
